service/prometheus/prometheus.go

```diff
@@ -16,6 +17,30 @@ const (
 	// ClusterIDLabel is the Prometheus label used to identify guest cluster
 	// metrics by external clients.
 	ClusterIDLabel = "cluster_id"
+
+	// NamespaceLabel is the Prometheus label we use internally for an
+	// endpoints namespace.
+	NameLabel = "kubernetes_name"
+
+	// NamespaceLabel is the Prometheus label we use internally for
+	// an endpoints namespace.
+	NamespaceLabel = "kubernetes_namespace"
+
+	// PrometheusNamespaceLabel is the Prometheus label added by the Kubernetes
+	// service discovery to hold an endpoint targets namespace.
+	PrometheusNamespaceLabel = "__meta_kubernetes_namespace"
+
+	// PrometheusServiceNameLabel is the Prometheus label added by the Kubernetes
+	// service discovery to hold an endpoints service name.
+	PrometheusServiceNameLabel = "__meta_kubernetes_service_name"
```

Comment:
Why the double underscore?

Reply:
This is a prometheus label, they chose it. I think they tend to prefix internal labels with double underscores.